### PR TITLE
ARCO-196: refactor errors and unify unit tests

### PR DESCRIPTION
**Pull request by pawellewandowski98**
## Description of Changes

First part of refactoring errors and unification unit tests

## Linked Issues / Tickets

ARCO-196

## Testing Procedure

Describe the tests you've added or any testing steps you've taken.

- [ ] I have added new unit tests
- [ ] All tests pass locally
- [ ] I have tested manually in my local environment

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have updated `CHANGELOG.md` with my changes

### CODE REVIEW

File: internal/woc_client/woc_client.go [hunk 1]
```diff
@@ -84,7 +94,7 @@ func (w *WocClient) GetUTXOsWithRetries(ctx context.Context, lockingScript *scri
 	operation := func() (sdkTx.UTXOs, error) {
 		wocUtxos, err := w.GetUTXOs(ctx, lockingScript, address)
 		if err != nil {
-			return nil, fmt.Errorf("failed to get utxos from WoC: %v", err)
+			return nil, errors.Join(ErrWOCFailedToGetUTXOs, fmt.Errorf("error: %v", err))
```
Comment (kuba-4chain) [inline internal/woc_client/woc_client.go:97]: This msg here is pointless, we can just do:
```suggestion
			return nil, errors.Join(ErrWOCFailedToGetUTXOs, err)
```

File: internal/woc_client/woc_client.go [hunk 2]
```diff
@@ -108,27 +118,23 @@ func (w *WocClient) GetUTXOs(ctx context.Context, lockingScript *script.Script,
...
 
-	if resp.StatusCode != 200 {
-		return nil, fmt.Errorf("response status not OK: %s", resp.Status)
-	}
-
 	var wocUnspent []*wocUtxo
 	err = json.NewDecoder(resp.Body).Decode(&wocUnspent)
 	if err != nil {
-		return nil, fmt.Errorf("failed to decode response: %v", err)
+		return nil, errors.Join(ErrWOCFailedToDecodeResponse, fmt.Errorf("error: %v", err))
```
Comment (kuba-4chain) [inline internal/woc_client/woc_client.go:130]:
```suggestion
		return nil, errors.Join(ErrWOCFailedToDecodeResponse, err)
```

File: internal/woc_client/woc_client.go [hunk 3]
```diff
@@ -108,27 +118,23 @@ func (w *WocClient) GetUTXOs(ctx context.Context, lockingScript *script.Script,
...
 	}
 
 	unspent := make(sdkTx.UTXOs, len(wocUnspent))
 	for i, utxo := range wocUnspent {
 		txIDBytes, err := hex.DecodeString(utxo.Txid)
 		if err != nil {
-			return nil, fmt.Errorf("failed to decode hex string: %v", err)
+			return nil, errors.Join(ErrWOCFailedToDecodeHexString, fmt.Errorf("error: %v", err))
```
Comment (kuba-4chain) [inline internal/woc_client/woc_client.go:137]:
```suggestion
			return nil, errors.Join(ErrWOCFailedToDecodeHexString, err)
```

File: internal/message_queue/nats/nats_connection/nats_connection.go [hunk 4]
```diff
@@ -49,7 +52,7 @@ func New(natsURL string, logger *slog.Logger) (*nats.Conn, error) {
 
 	nc, err = nats.Connect(natsURL, opts...)
 	if err != nil {
-		return nil, fmt.Errorf("failed to connect to NATS server: %v", err)
+		return nil, errors.Join(ErrNatsConnectionFailed, fmt.Errorf("error: %v", err))
```
Comment (kuba-4chain) [inline internal/message_queue/nats/nats_connection/nats_connection.go:55]:
```suggestion
		return nil, errors.Join(ErrNatsConnectionFailed, err)
```

File: internal/woc_client/woc_client.go [hunk 5]
```diff
@@ -148,20 +154,16 @@ func (w *WocClient) GetBalance(ctx context.Context, address string) (int64, int6
...
 
-	if resp.StatusCode != 200 {
-		return 0, 0, fmt.Errorf("response status not OK: %s", resp.Status)
-	}
-
 	var balance wocBalance
 	err = json.NewDecoder(resp.Body).Decode(&balance)
 	if err != nil {
-		return 0, 0, fmt.Errorf("failed to decode response: %v", err)
+		return 0, 0, errors.Join(ErrWOCFailedToDecodeResponse, fmt.Errorf("error: %v", err))
```
Comment (kuba-4chain) [inline internal/woc_client/woc_client.go:166]:
```suggestion
		return 0, 0, errors.Join(ErrWOCFailedToDecodeResponse, err)
```

File: internal/woc_client/woc_client.go [hunk 6]
```diff
@@ -279,7 +272,7 @@ func (w WocClient) httpRequest(ctx context.Context, method string, endpoint stri
 
 	req, err := http.NewRequestWithContext(ctx, method, fmt.Sprintf("%s/%s/%s", apiUrl, w.net, endpoint), body)
 	if err != nil {
-		return nil, fmt.Errorf("failed to crreate request: %v", err)
+		return nil, errors.Join(ErrWOCFailedToCreateRequest, fmt.Errorf("error: %v", err))
```
Comment (kuba-4chain) [inline internal/woc_client/woc_client.go:275]:
```suggestion
		return nil, errors.Join(ErrWOCFailedToCreateRequest, err)
```

File: internal/woc_client/woc_client.go [hunk 7]
```diff
@@ -288,3 +281,16 @@ func (w WocClient) httpRequest(ctx context.Context, method string, endpoint stri
 
 	return req, nil
 }
+
+func (w *WocClient) doRequest(req *http.Request) (*http.Response, error) {
+	resp, err := w.client.Do(req)
+	if err != nil {
+		return nil, errors.Join(ErrWOCRequestFailed, fmt.Errorf("error: %v", err))
```
Comment (kuba-4chain) [inline internal/woc_client/woc_client.go:288]:
```suggestion
		return nil, errors.Join(ErrWOCRequestFailed, err)
```

File: internal/woc_client/woc_client.go (hunk 1 above)
Comment (pawellewandowski98) [reply internal/woc_client/woc_client.go]: Done

File: internal/woc_client/woc_client.go (hunk 2 above)
Comment (pawellewandowski98) [reply internal/woc_client/woc_client.go]: Done

File: internal/woc_client/woc_client.go (hunk 3 above)
Comment (pawellewandowski98) [reply internal/woc_client/woc_client.go]: Done

File: internal/message_queue/nats/nats_connection/nats_connection.go (hunk 4 above)
Comment (pawellewandowski98) [reply internal/message_queue/nats/nats_connection/nats_connection.go:55]: Done

File: internal/woc_client/woc_client.go (hunk 5 above)
Comment (pawellewandowski98) [reply internal/woc_client/woc_client.go]: Done

File: internal/woc_client/woc_client.go (hunk 6 above)
Comment (pawellewandowski98) [reply internal/woc_client/woc_client.go]: Done

File: internal/woc_client/woc_client.go (hunk 7 above)
Comment (pawellewandowski98) [reply internal/woc_client/woc_client.go]: Done

File: internal/beef/beef_test.go [hunk 8]
```diff
@@ -103,109 +106,110 @@ func TestDecodeBEEF_DecodeBEEF_HappyPaths(t *testing.T) {
...
 	bumpError := errors.New("BUMP bytes do not contain enough data to be valid")
 	varIntError := errors.New("could not read varint type: EOF")
+
 	testCases := []struct {
 		name                         string
 		hexStream                    string
 		expectedDecodedBEEF          *BEEF
 		expectedCMPForTheOldestInput bool
 		expectedError                error
+		expectedExternalError        error
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply internal/beef/beef_test.go]: Done

**Review by boecklim — APPROVED** [review]

**Pull request merged**